lib/analyzer/image-extractor.ts

```diff
@@ -0,0 +1,198 @@
+import { createReadStream } from "fs";
```

Comment:
this will also be a good opportunity to organise this file a bit and turn it into a dir -
move the types to `types.ts`
things we want to export to `index.ts`
if it grows too big, extract things to dedicated files, etc
what do you think?

Reply:
Just to note why this is resolved: we had a conversation about the organisation of the files and I'll update the PR with the new proposed layout